### PR TITLE
Adding custom fields selection

### PR DESCRIPTION
**Pull request by rafiki270**
Requires to be tagged first:
https://github.com/vapor/database-kit/pull/20

### CODE REVIEW

File: Sources/Fluent/Query/QueryField.swift [hunk 1]
```diff
@@ -163,3 +163,51 @@ public struct QueryFieldEncodingContainer<Model: Fluent.Model> {
         try container.encode(value, forKey: field)
     }
 }
+
+/// Creating select all field
+extension QueryField {
+    /// Predefined select all field
+    static func all() -> QueryField {
+        return QueryField(name: "*")
```
Comment (tanner0101) [inline Sources/Fluent/Query/QueryField.swift]: The concept of `*` is SQL-specific. We need something generic.

Comment (rafiki270) [reply Sources/Fluent/Query/QueryField.swift]: I was giving this only a small chance of success ... I'll remove it :)

**Comment by rafiki270** [comment]
@tanner0101 I was thinking I would add something like a `FieldRepresentable` which then could take custom sub queries like `SELECT *, (SELECT COUNT(id) FROM table-b) as count FROM table-a` (obviously in fluent you would be passing in another query which only selects one column) ... it would also could take enums (or fake enums from all the drivers like `.concat(fieldOrString: FieldRepresentable, ...)` or `.year(\MyModel.column) ... etc ... please let me know if this would be a solution you'd be happy to go with ...

**Comment by tanner0101** [comment]
I think that might make more sense to do with `customSQL` or using the database connection's methods manually. The custom payload generated by that won't be very useful for parsing a model from anyway, which is Fluent's primary purpose.

**Comment by tanner0101** [comment]
This is possible now using `builder.query.keys`. Let me know if you have any questions!

**Pull request closed without merging**